### PR TITLE
cs/cas-414 Import Announcement Subscribers

### PR DESCRIPTION
**Pull request by CorySanin**
import subscribers from a csv generated through the announcements panel. This is an extension to the functionality added by #222.

### CODE REVIEW

**Comment by tkimnguyen** [comment]
Please add some documentation for all changed and new features!

**Comment by CorySanin** [comment]
Will do. Adding WIP tag for now.

File: castle/cms/browser/controlpanel/announcements.py [hunk 1]
```diff
@@ -255,7 +259,7 @@ def handle_export(self, action):
             responsebody = ','.join(fields)
             categories = set()
             if 'form.widgets.export_categories' in self.request.form:
-                if data['export_categories'] != u'':
+                if data['export_categories'] is not None and data['export_categories'] != u'':
```
Comment (vangheem) [inline castle/cms/browser/controlpanel/announcements.py]: Maybe better to just rewrite to `if data['export_categories']:` or if you really want to be explicit, `if data['export_categories'] not in (None, ''):`

File: castle/cms/browser/controlpanel/announcements.py [hunk 2]
```diff
@@ -303,7 +363,7 @@ def handle_merge(self, action):
             categories = set()
             newname = u''
             if 'form.widgets.rename_merge_categories' in self.request.form:
-                if data['rename_merge_categories'] != u'':
+                if data['rename_merge_categories'] is not None and data['rename_merge_categories'] != u'':
```
Comment (vangheem) [inline castle/cms/browser/controlpanel/announcements.py]: same here

File: castle/cms/browser/controlpanel/announcements.py [hunk 3]
```diff
@@ -361,7 +421,7 @@ def handle_delete(self, action):
             allcategories = api.portal.get_registry_record(reg_key)
             categories = set()
             if 'form.widgets.delete_categories' in self.request.form:
-                if data['delete_categories'] != u'':
+                if data['delete_categories'] is not None and data['delete_categories'] != u'':
```
Comment (vangheem) [inline castle/cms/browser/controlpanel/announcements.py]: yup

File: castle/cms/browser/controlpanel/announcements.py [hunk 4]
```diff
@@ -414,7 +474,7 @@ def handle_addcat(self, action):
             allcategories = api.portal.get_registry_record(reg_key)
             categories = set()
             if 'add_categories' in data:
-                if data['add_categories'] != u'':
+                if data['add_categories'] is not None and data['add_categories'] != u'':
```
Comment (vangheem) [inline castle/cms/browser/controlpanel/announcements.py]: ^^, sorry

File: castle/cms/browser/controlpanel/announcements.py (hunk 1 above)
Comment (CorySanin) [reply castle/cms/browser/controlpanel/announcements.py]: Will do

**Comment by CorySanin** [comment]
uhh I think the rebase worked

**Comment by CorySanin** [comment]
closing PR due to github messing up.

**Pull request closed without merging**